src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Write.cs

```diff
@@ -374,12 +374,9 @@ private int WritePosixName(Span<byte> buffer)
 
             if (_name.Length > FieldLengths.Name)
             {
-                int prefixBytesLength = Math.Min(_name.Length - FieldLengths.Name, FieldLengths.Name);
-                Span<byte> remaining = prefixBytesLength <= 256 ?
-                    stackalloc byte[prefixBytesLength] :
-                    new byte[prefixBytesLength];
-
-                int encoded = Encoding.ASCII.GetBytes(_name.AsSpan(FieldLengths.Name), remaining);
+                int prefixBytesLength = Math.Min(_name.Length - FieldLengths.Name, FieldLengths.Prefix);
+                Span<byte> remaining = stackalloc byte[prefixBytesLength];
+                int encoded = Encoding.ASCII.GetBytes(_name.AsSpan(FieldLengths.Name, prefixBytesLength), remaining);
```

Comment:
Do we have any tests for what happens if non-ASCII values are used?

Reply:
Seems like we don't. Only pax works as expected and that's because it replaces the name with the extended attribute "path", which uses UTF8 on write and read, other formats like GNU and V7 write and read using ASCII. 

For said formats, the name is garbled. Not sure if we should validate non-ASCII names and if there's a way of doing it. But the ASCII usage seems intentional, perhaps that's what the spec says. @carlossanlop might know better.
```
      System.Formats.Tar.Tests.TarWriter_WriteEntry_Tests.WriteLongUtf8Name(format: Gnu) [FAIL]
        Assert.Equal() Failure
                  ↓ (pos 0)
        Expected: ùöüüoüüüyúü÷ù_úÿ_û_yû_oÿÿyùyûyyöû÷_öúÿoû_···
        Actual:   ?????????????????????????????????????????···
```

Reply:
https://github.com/dotnet/runtime/issues/75482